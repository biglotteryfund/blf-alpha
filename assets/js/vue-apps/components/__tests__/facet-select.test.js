/**
 * @jest-environment jsdom
 */
/* eslint-env jest */
import { mount } from '@vue/test-utils';
import FacetSelect from '../facet-select.vue';

describe('FacetSelect', () => {
    it('should render with plain options', () => {
        const wrapper = mount(FacetSelect, {
            propsData: {
                name: 'example',
                label: 'Example',
                clearLabel: 'Clear selection',
                value: 'option-1',
                options: [
                    {
                        label: 'Option 1',
                        value: 'option-1'
                    },
                    {
                        label: 'Option 2',
                        value: 'option-2'
                    }
                ]
            }
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it('should render with optgroup', () => {
        const wrapper = mount(FacetSelect, {
            propsData: {
                name: 'example',
                label: 'Example',
                clearLabel: 'Clear selection',
                options: {
                    'Group 1': [
                        {
                            label: 'Option 1',
                            value: 'option-1'
                        },
                        {
                            label: 'Option 2',
                            value: 'option-2'
                        }
                    ],
                    'Group 2': [
                        {
                            label: 'Option 3',
                            value: 'option-2'
                        },
                        {
                            label: 'Option 4',
                            value: 'option-4'
                        }
                    ]
                }
            }
        });

        expect(wrapper.element).toMatchSnapshot();
    });
});
