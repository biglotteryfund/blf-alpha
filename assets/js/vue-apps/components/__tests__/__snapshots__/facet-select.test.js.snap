// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacetSelect should render with optgroup 1`] = `
<div>
  <label
    class="ff-label"
    for="field-dynamic-example"
  >
     Example 
  </label>
   
  <div
    class="ff-choice__option ff-choice__option--flex"
  >
    <select
      class="ff-select"
      id="field-dynamic-example"
      name="example"
    >
      <!---->
       
      <optgroup
        label="Group 1"
      >
        <option
          value="option-1"
        >
          
                        Option 1
                    
        </option>
        <option
          value="option-2"
        >
          
                        Option 2
                    
        </option>
      </optgroup>
      <optgroup
        label="Group 2"
      >
        <option
          value="option-2"
        >
          
                        Option 3
                    
        </option>
        <option
          value="option-4"
        >
          
                        Option 4
                    
        </option>
      </optgroup>
    </select>
     
    <!---->
  </div>
</div>
`;

exports[`FacetSelect should render with plain options 1`] = `
<div>
  <label
    class="ff-label"
    for="field-dynamic-example"
  >
     Example 
  </label>
   
  <div
    class="ff-choice__option ff-choice__option--flex"
  >
    <select
      class="ff-select"
      id="field-dynamic-example"
      name="example"
    >
      <!---->
       
      <option
        value="option-1"
      >
        
                    Option 1
                
      </option>
      <option
        value="option-2"
      >
        
                    Option 2
                
      </option>
    </select>
     
    <button
      class="active-filter active-filter--mini u-margin-left-s"
    >
      <svg
        aria-describedby="aria-close-desc-clear-field-dynamic-example"
        class="icon icon--close"
        role="img"
        viewBox="0 0 512 512"
        width="46"
        xmlns="http://www.w3.org/2000/svg"
      >
        <desc
          id="aria-close-desc-clear-field-dynamic-example"
        >
          Clear selection
        </desc>
         
        <path
          d="M340.2 160l-84.4 84.3-84-83.9-11.8 11.8 84 83.8-84 83.9 11.8 11.7 84-83.8 84.4 84.2 11.8-11.7-84.4-84.3 84.4-84.2z"
        />
      </svg>
    </button>
  </div>
</div>
`;
